lib/streamlit/string_util.py

```diff
@@ -67,6 +73,46 @@ def validate_emoji(maybe_emoji: str | None) -> str:
         )
 
 
+def validate_material_icon(maybe_material_icon: str | None) -> str:
+
+    supported_icon_packs = [
+        "material",
+    ]
+
+    if maybe_material_icon is None:
+        return ""
+
+    icon_regex = r"^\s*:(.*)\/(.*):\s*$"
+    icon_match = re.match(icon_regex, maybe_material_icon)
+
+    if not icon_match:
+        raise StreamlitAPIException(
+            f'The value `"{maybe_material_icon}"` is not a valid Material icon. '
+            f"Please use a Material icon shortcode like **`:material/thumb_up:`**"
+        )
+    pack_name, icon_name = icon_match.groups()
+
+    if pack_name not in supported_icon_packs:
+        icon_packs_repr = ", ".join(f"`{pack}`" for pack in supported_icon_packs)
+        raise StreamlitAPIException(
+            f'The value `"{maybe_material_icon}"` is not a valid Material icon.'
+            f" Please use a Material icon shortcode like **`:material/thumb_up:`**. "
+            f"Only {icon_packs_repr} icon packs are supported."
+        )
+    if not icon_name:
+        raise StreamlitAPIException(
+            f'The value `"{maybe_material_icon}"` is not a valid Material icon.'
+            f" Please use a Material icon shortcode, like **`:material/thumb_up:`**"
+        )
+
+    if not is_material_icon(icon_name):
+        raise StreamlitAPIException(
+            f'The value `"{maybe_material_icon}"` is not a valid Material icon.'
+            f" Please use a Material icon shortcode, like **`:material/thumb_up:`**"
+        )
```

Comment:
nit: can these `if` statements be combined?

Reply:
Hm, also minor, but could we generalize the error message construction here since 3 of them seem to be the same, and the only difference in the fourth is a suffix